### PR TITLE
fix annoying 'please check...' messages; remove v1beta1&2 ref

### PR DESCRIPTION
**Pull request by lavalamp**
@bgrant0607 @caesarxuchao @krousey 

Part of this is a thing that was missed in removing v1beta1&2, and also in adding v1.

### CODE REVIEW

**Comment by bgrant0607** [comment]
Thanks. This was listed in #8087, but wasn't prioritized correctly.

**Comment by caesarxuchao** [comment]
#9854 tries to add the fieldmapping for v1, but this PR seems more comprehensive.

File: pkg/client/request.go [hunk 1]
```diff
@@ -302,45 +312,41 @@ func (v versionToResourceToFieldMapping) filterField(apiVersion, resourceType, f
...
+			EventInvolvedNamespace:       "involvedObject.namespace",
+			EventInvolvedName:            "involvedObject.name",
+			EventInvolvedUID:             "involvedObject.uid",
+			EventInvolvedAPIVersion:      "involvedObject.apiVersion",
+			EventInvolvedResourceVersion: "involvedObject.resourceVersion",
+			EventInvolvedFieldPath:       "involvedObject.fieldPath",
 		},
 	},
-	"v1beta3": resourceTypeToFieldMapping{
+	"v1": resourceTypeToFieldMapping{
```
Comment (bgrant0607) [inline pkg/client/request.go]: PodHost is spec.nodeName in v1.

**Comment by bgrant0607** [comment]
cc @liggitt

File: pkg/client/request.go [hunk 2]
```diff
@@ -302,45 +312,41 @@ func (v versionToResourceToFieldMapping) filterField(apiVersion, resourceType, f
...
 		"nodes": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
+			ObjectNameField:   "metadata.name",
+			NodeUnschedulable: "spec.unschedulable",
 		},
 		"minions": clientFieldNameToAPIVersionFieldName{
-			ObjectNameField:   "name",
-			NodeUnschedulable: "unschedulable",
+			ObjectNameField:   "metadata.name",
```
Comment (bgrant0607) [inline pkg/client/request.go]: We should eliminate minions, at least in v1.

**Comment by bgrant0607** [comment]
Tagged with v1.0 because it's part of #7018 that was missed.

File: pkg/client/request.go (hunk 1 above)
Comment (liggitt) [inline pkg/client/request.go]: that may be true, but this needs updating as well then:
```
func PodToSelectableFields(pod *api.Pod) fields.Set {
	return fields.Set{
		"metadata.name": pod.Name,
		"spec.host":     pod.Spec.NodeName,
		"status.phase":  string(pod.Status.Phase),
	}
}
```

Comment (liggitt) [inline pkg/client/request.go]: if I understand correctly, all of these things have to match:
* the label returned from the `AddFieldLabelConversionFunc` for Pod in all versions
* the key returned from `PodToSelectableFields`
* the value of `client.PodHost`

Comment (bgrant0607) [inline pkg/client/request.go]: Good point. Yes, we should fix those.

**Pull request merged**